EIPS/eip-7727.md

```diff
@@ -0,0 +1,228 @@
+
+## Motivation
+
+Currently, a single block builder has unrestricted control over the final sequencing of a block’s transactions. This poses a problem, as sequencing—the choice of who gets to interact with specific pieces of state and in what order—significantly influences value flow. The objective of this EIP is to allow more parties to participate in the construction of single blocks by exposing sequencing concepts inside of the EVM. This change would enable EVM applications to reclaim some of the sequencing value that is currently monopolized by block builders, redirecting it back to the applications themselves.
+
+## Specification
+
+### Constants
+
+| Name | Value |
+| --- | --- |
+| FORK_BLOCK | TBD |
+| DELEGATED_TX_TYPE | 0x05 |
+| BUNDLE_TX_TYPE | 0x06 |
+| BUNDLE_BASE_GAS_COST | TBD |
+| BUNDLE_SIGNER_OPCODE_COST | TBD |
+| BUNDLE_SIGNER_OPCODE_NUMBER | TBD |
+
+### New Transaction Payload Types
+
+Two new [EIP-2718](./eip-2718.md) transactions with types `DELEGATED_TX_TYPE` and `BUNDLE_TX_TYPE` are introduced at `FORK_BLOCK`.
+
+For the `DELEGATED_TX_TYPE`, the transaction payload should be interpreted as:
+
+```go
+0x05 || rlp([
+  bundleSigner,
+  blockNumber,
+	chainId, 
+	nonce, 
+	gasPrice, 
+	gasLimit, 
+	to, 
+	value, 
+	data, 
+	signatureYParity, 
+	signatureR, 
+	signatureS
+])
+```
+
+The `signatureYParity`, `signatureR`, `signatureS` elements of the `DELEGATED_TX_TYPE` represent a secp256k1 signature over:
+
+```go
+keccak256(0x05 || rlp([bundleSigner, blockNumber, chain_id, nonce, gasPrice, gasLimit, to, value, data]))
+```
+
+For the `BUNDLE_TX_TYPE`, the transaction payload should be interpreted as:
+
+```go
+0x06 || rlp([
+  bundleSigner,
+  blockNumber,
+	chainId, 
+	nonce, 
+	gasPrice, 
+	transactionList
+	signatureYParity, 
+	signatureR, 
+	signatureS
+])
+```
+
+Where the `transactionList` element is a list of syntactically valid transactions of either type `DELEGATED_TX_TYPE` or `BUNDLE_TX_TYPE`. At least one transaction must be in the list.
+
+An example would of the `transactionList` would be:
+
+```go
+[
+	DELEGATED_TX_TYPE.payload, 
+	BUNDLE_TX_TYPE.payload, 
+	DELEGATED_TX_TYPE.payload
+]
+```
+
+The `signatureYParity`, `signatureR`, `signatureS` elements of the `BUNDLE_TX_TYPE` represent a secp256k1 signature over:
+
+```go
+keccak256(0x06 || rlp([bundleSigner, blockNumber, chainId, nonce, gasPrice, transactionList]))
+```
+
+We’ll refer to address resolved by this signature the bundle transaction’s signer. 
+
+### `BUNDLE_SIGNER` Opcode
+
+The `BUNDLE_SIGNER` is a new opcode, added in `FORK_BLOCK`, identified by `BUNDLE_SIGNER_OPCODE_NUMBER` and requires zero stack arguments.
+
+When the transaction type is `DELEGATED_TX_TYPE`, this opcode pushes the `bundleSigner` from the transaction payload onto the stack as an address. If the transaction is of a different type, it returns the zero address.
+
+The gas cost for this opcode is `BUNDLE_SIGNER_OPCODE_COST`.
```

Comment:
If you use one of the constants from EELS, for example [`GAS_VERY_LOW`](https://github.com/ethereum/execution-specs/blob/7df0ecda3274b72b42fdf02afb077fc74fa14fd4/src/ethereum/arrow_glacier/vm/gas.py#L26), it helps keep things a bit more organized. This is not a requirement to get merged, just a suggestion.